gen/app-gen

```diff
@@ -0,0 +1,307 @@
+#!/bin/bash
+
+set -e
+
+show_help() {
+    cat << EOF
+
+Simple tool to generate Mender Artifact suitable for App Update Module
+
+Usage: $0 [options] [-- [options-for-mender-artifact] ]
+
+    Options: [ -n|--artifact-name -t|--device-type -o|--output_path --i|--image -p|--platform -r|--orchestrator -m|--manifests-dir -h|--help -a|--application-name ]
+
+        --artifact-name     - Artifact name
+        --device-type       - Target device type identification (can be given more than once)
+        --output-path       - Path to output artifact file. Default: reboot-artifact.mender
+        --image [current-url,]new-url       - Path to output artifact file. Default: reboot-artifact.mender
+        --manifests-dir     - Directory containing orchestrator-specific manifests describing the deployment
+        --platform          - Platform of the images, e.g.: linux/arm/v7
+        --orchestrator      - Name of the orchestrator, and name of the sub-module, e.g.: docker-compose
+        --application-name  - Name of the application running on a device
+        --help              - Show help and exit
+
+Anything after a '--' gets passed directly to the mender-artifact tool.
+
+EOF
+}
+
+show_help_and_exit_error() {
+    show_help
+    exit 1
+}
+
+check_dependency() {
+    if ! which "$1" > /dev/null; then
+        echo "The $1 utility is not found but required to generate Artifacts." 1>&2
+        return 1
+    fi
+}
+
+if ! check_dependency mender-artifact; then
+    echo "Please follow the instructions here to install mender-artifact and then try again: https://docs.mender.io/downloads#mender-artifact" 1>&2
+    exit 1
+fi
+
+delta_cmd="xdelta3 -e -s"
+device_types=""
+artifact_name=""
+output_path="app-artifact.mender"
+passthrough=0
+passthrough_args=""
+version="1.0"
+orchestrator=""
+platform=""
+manifests_dir=""
+declare -a images
+declare -a images_sha
+declare -a images_ids
+
+while [ -n "$1" ]; do
+    if test $passthrough -eq 1; then
+        passthrough_args="$passthrough_args $1"
+        shift
+        continue
+    fi
+    case "$1" in
+        --application-name | -a)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            if [[ $application_name =~ [^a-zA-Z0-9_-] ]]; then
+                echo "ERROR: application name must contain only alpha-numerics, _ or -"
+                show_help_and_exit_error
+            fi
+            application_name="$2"
+            shift 2
+            ;;
+        --manifests-dir | -r)
```

Comment:
`-r` used both here and with --orchestrator

Reply:
it was supposed to be -m (the help screen was correct, only small ;-) detail in the code). sorry. thanks. fixed.